### PR TITLE
Use matplotlib for better colormaps

### PR DESCRIPTION
**Pull request by lukeyeager**
Adds dependency on `matplotlib` (pretty standard package)
Use their default colormap - "viridis"

http://bids.github.io/colormap/

Jet | Viridis | Magma

heatmap-options

### CODE REVIEW

File: digits/utils/image.py [hunk 1]
```diff
@@ -443,17 +446,21 @@ def vis_square(images,
...
-        red = np.interp(images*(len(redmap)-1)/255.0, xrange(len(redmap)), redmap)
-        green = np.interp(images*(len(greenmap)-1)/255.0, xrange(len(greenmap)), greenmap)
-        blue = np.interp(images*(len(bluemap)-1)/255.0, xrange(len(bluemap)), bluemap)
+        # they're grayscale - convert to color
+        color_images = np.ndarray(images.shape + (3,), dtype=images.dtype)
+        if colormap is None:
+            try:
+                converter = matplotlib.cm.viridis
+            except AttributeError:
+                # old verison of matplotlib
```
Comment (gheinrich) [inline digits/utils/image.py]: small typo

**Comment by gheinrich** [comment]
It is taking much longer to process data now. To visualize DetectNet layers: 2m37s with this PR, 6s without. Are you seeing the same? I have Matplotlib 1.5.1.

**Comment by lukeyeager** [comment]
Oh gosh, yes you're right. Visualizing AlexNet goes from 3s to 30s. Forget that.

**Comment by thatguymike** [comment]
Seems like a problem...

**Pull request closed without merging**